### PR TITLE
fix(43939): Quick fix shouldn't be available for converting getters to async await

### LINKED ISSUE #43939: Quick fix shouldn't be available for converting getters to async await
# Bug Report

getters cannot be async, this quick fix results in a compile error:

Screen Shot 2021-05-04 at 12 07 54 AM

Screen Shot 2021-05-04 at 12 11 27 AM

### 🔎 Search Terms

async, get, property

### 🕗 Version & Regression Information

v4.3.0-dev.20120426

### ⏯ Playground Link

Doesn't seem to happen here, needs TS extension in vscode

### 💻 Code

This was the code:

```ts
	get linuxDistro(): Promise<LinuxDistro> {
		if (!isLinux) {
			return Promise.resolve(LinuxDistro.Unknown);
		}
		const file = '/etc/os-release';
		return SymlinkSupport.existsFile(file).then(async exists => {
			if (!exists) {
				return LinuxDistro.Unknown;
			}
			const buffer = await fsPromises.readFile(file);
			const contents = buffer.toString();
			if (/NAME="?Fedora"?/.test(contents)) {
				return LinuxDistro.Fedora;
			} else if (/NAME="?Ubuntu"?/.test(contents)) {
				return LinuxDistro.Ubuntu;
			} else {
				return LinuxDistro.Unknown;
			}
		});
	}
```

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #43939

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Doesn't this also allow setters? I would just limit it to arrow functions, function expressions, declarations, and method declarations.

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser oke

File: src/services/suggestionDiagnostics.ts [hunk 1]
```diff
@@ -213,4 +213,16 @@ namespace ts {
 
         return false;
     }
+
+    export function isAsyncLikeFunction(node: Node): node is FunctionDeclaration | MethodDeclaration | FunctionExpression | ArrowFunction {
```
Comment (DanielRosenwasser) [inline src/services/suggestionDiagnostics.ts:217]: I would potentially rename this to `canBeConvertedToAsync`

```suggestion
    export function isFunctionThatSupportsAsync(node: Node): node is FunctionDeclaration | MethodDeclaration | FunctionExpression | ArrowFunction {
```

File: src/services/suggestionDiagnostics.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ namespace ts {
                 }
             }
 
-            if (isFunctionLikeDeclaration(node)) {
+            if (isAsyncLikeFunction(node)) {
```
Comment (DanielRosenwasser) [inline src/services/suggestionDiagnostics.ts:60]:
```suggestion
            if (isFunctionThatSupportsAsync(node)) {
```

File: src/services/codefixes/convertToAsyncFunction.ts [hunk 3]
```diff
@@ -54,7 +54,7 @@ namespace ts.codefix {
             functionToConvert = tokenAtPosition.parent.initializer;
         }
         else {
-            functionToConvert = tryCast(getContainingFunction(getTokenAtPosition(sourceFile, position)), isFunctionLikeDeclaration);
+            functionToConvert = tryCast(getContainingFunction(getTokenAtPosition(sourceFile, position)), isAsyncLikeFunction);
```
Comment (DanielRosenwasser) [inline src/services/codefixes/convertToAsyncFunction.ts:57]:
```suggestion
            functionToConvert = tryCast(getContainingFunction(getTokenAtPosition(sourceFile, position)), isFunctionThatSupportsAsync);
```

**Review by DanielRosenwasser — APPROVED** [review]

File: src/services/suggestionDiagnostics.ts (hunk 1 above)
Comment (a-tarasyuk) [reply src/services/suggestionDiagnostics.ts]: Renamed to `canBeConvertedToAsync`

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**